metricflow/model/parsing/dir_to_model.py

```diff
@@ -82,13 +84,21 @@ def parse_directory_of_yaml_files_to_model(
     model = build_result.model
     assert model
 
-    if apply_pre_transformations:
-        model = ModelTransformer.pre_validation_transform_model(model)
+    build_issues = build_result.issues
+    try:
+        if apply_pre_transformations:
+            model = ModelTransformer.pre_validation_transform_model(model)
+
+        if apply_post_transformations:
+            model = ModelTransformer.post_validation_transform_model(model)
+    except Exception as e:
+        transformation_issue_results = ModelValidationResults(errors=[ValidationError(message=str(e))])
+        build_issues = ModelValidationResults.merge([build_issues, transformation_issue_results])
 
-    if apply_post_transformations:
-        model = ModelTransformer.post_validation_transform_model(model)
+    if raise_issues_as_exceptions and build_issues.has_blocking_issues:
+        raise ModelValidationException(build_issues.all_issues)
```

Comment:
So if we can store the original exceptions we can throw this from those, and we'd be able to collect all of the stack traces. Here it just has string pointers to things happening... somewhere.

I've had mixed experiences with this kind of global "collect issues and maybe throw later" implementation pattern - I think we might want to have cleaner separation between what constitutes an exception and what constitutes a user error, and throw exceptions for the former and collect issues for the latter. I'm not sure how to go about achieving this, honestly, because I don't know where our parsing error boundary ends and the user input error boundary begins, but it's something to think about as we tinker further here.

In the meantime, preserving stack trace information is probably sufficient.